### ISSUE TITLE
Checking if the url is an audio stream ?

### ISSUE THREAD

**Issue by satyres** [Alamofire/Alamofire#3100]
Hi
i've been trying to find out if the current Url is a live stream or (PLS , m3u) file that contain the direct stream , but AF return nothing when it's direct stream and it kinda freeze.
here is the code
```
 AF.request(streamUrl)
        .validate(statusCode: 200..<300)
    
        .responseString { response in
            print(response.response?.allHeaderFields)
            switch response.result {
                
            case .success:
                
                print("response2 \(response.value) ")
                let detector = try! NSDataDetector(types: NSTextCheckingResult.CheckingType.link.rawValue)
                let matches = detector.matches(in: response.value!, options: [], range: NSRange(location: 0, length: response.value!.utf16.count))

                for match in matches {
                    guard let range = Range(match.range, in: response.value!) else { continue }
                    let url = response.value![range]
                    print("direct=\(url)")
                }
                
                
                
            case let .failure(error):
                print("response3 \(error)")
            }
        } 
```
please , is there a way to solve this issue.
Regards

**Comment by jshier** [comment]
Alamofire doesn't currently handle streaming `Data`, so that's likely why you're seeing what appears to be a hang: the `Data` is streaming into memory and never being completed. However, we are working on a `DataStreamRequest` which should do what you want. It would be great if you could try it out on the `feature/data-streaming` branch and let us know if it works for you. It would look something like this:

```swift
AF.streamRequest(streamUrl)
    .validate()
    .responseStream { output in
        // output.value should have the incoming Data
    }
```

Give it a try and let us know!

**Comment by satyres** [comment]
Hi
Thanks so much for your help.
i've updated to the last version 5.0.2 and i still have problem streamRequest is unkown !
**Value of type 'Session' has no member 'streamRequest'**
what am i missing ?
regards

**Comment by jshier** [comment]
You need to use our `feature/data-streaming` branch to preview the streaming feature.

**Comment by satyres** [comment]
Hi
Thanks so much for this new feature , yes i confirm it's working fine and showing how much bytes have been read . this is exactly what i want Thanks.
Regards

**Comment by satyres** [comment]
Hi 
last question please how can i stop **responseStream** 
regards

**Comment by jshier** [comment]
You can capture the request when you create it and then call `cancel()` when you're done. Alternatively, you can throw an error in `responseStream` which will cancel for you and set the request's `error` property to that value. We're still trying to decide how this will work.

**Comment by satyres** [comment]
Great thanks so much ! very good feature , very helpful

**Comment by satyres** [comment]
Hi 
i'm trying to catch the error in case the url stream is offline but i couldn"t with 
```
if output.error != nil
                {
                    print("error getting link")
                    throw "Some Error"
                }
```
Thanks

**Comment by jshier** [comment]
You'll need to switch over the full output if you want to capture the `.complete` event, which is fired when the connection is closed.

**Comment by jshier** [comment]
@satyres I've just pushed a major refactor to the branch. Now, instead of taking the enum directly in the closure, a new `Stream` type is received, with an `event` property that is the previous `Output` type (renamed `Event`). I did this to enable inline cancellation, so you can now cancel the stream directly in the closure:

```swift
AF.streamRequest(...).responseStream { stream in
    switch stream.event {
    // Previous logic
    }

     // To cancel
    stream.cancel()
}
```

Cancellation will also occur if an `Error` is thrown from the stream closure, or, if you pass `automaticallyCancelOnStreamError` when creating the request, when a serializer hits an error.

Feel free to give it a try!

**Comment by satyres** [comment]
@jshier , Thanks so much for your effort and time.
i'll give a it a try and provide you my feedback.
Best regards

**Comment by jshier** [comment]
`DataStreamRequest` has now been released as part of Alamofire 5.1.
